client/post-editor/media-modal/secondary-actions.jsx

```diff
@@ -94,42 +72,24 @@ class MediaModalSecondaryActions extends Component {
 			<div>
 				{ this.getButtons().map( ( button ) => (
 					<Button
+						key={ button.key }
 						className={ classNames( 'editor-media-modal__secondary-action', button.className ) }
 						data-e2e-button={ button.key }
 						compact
-						{ ...pick( button, [ 'key', 'disabled', 'onClick', 'primary' ] ) }
+						disabled={ button.disabled }
+						onClick={ button.onClick }
+						primary={ button.primary }
 					>
 						{ button.icon && <Gridicon icon={ button.icon } /> }
-						{ button.text && button.text }
+						{ button.text }
 					</Button>
 				) ) }
 			</div>
 		);
 	}
 }
 
-export default connect(
-	( state, ownProps ) => ( {
-		view: getMediaModalView( state ),
-		user: getCurrentUser( state ),
-		siteSlug: ownProps.site ? getSiteSlug( state, ownProps.site.ID ) : '',
-		hideButton: ! canCurrentUser( state, ownProps.site.ID, 'publish_posts' ),
-	} ),
-	{
-		onViewDetails: flow(
-			withAnalytics( bumpStat( 'editor_media_actions', 'edit_button_dialog' ) ),
-			withAnalytics( recordGoogleEvent( 'Media', 'Clicked Dialog Edit Button' ) ),
-			partial( setEditorMediaModalView, ModalViews.DETAIL )
-		),
-	},
-	function mergeProps( stateProps, dispatchProps, ownProps ) {
-		//We want to overwrite connected props if 'onViewDetails', 'view' were provided
-		return Object.assign(
-			{},
-			ownProps,
-			stateProps,
-			dispatchProps,
-			pick( ownProps, [ 'onViewDetails', 'view' ] )
-		);
-	}
-)( localize( MediaModalSecondaryActions ) );
+export default connect( ( state, ownProps ) => ( {
+	user: getCurrentUser( state ),
+	hideButton: ! canCurrentUser( state, ownProps.site.ID, 'publish_posts' ),
```

Comment:
We could simplify by destructuring `ownProps` here, because we only need `site` from it:

```suggestion
export default connect( ( state, { site } ) => ( {
	user: getCurrentUser( state ),
	hideButton: ! canCurrentUser( state, site.ID, 'publish_posts' ),
```